### PR TITLE
Various updates to fix CVEs

### PR DESCRIPTION
**Pull request by dfarrell07**
See commit messages for details.

### CODE REVIEW

File: tools/go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/submariner-io/shipyard/tools
 
-go 1.20
+go 1.21
+
+toolchain go1.21.3
```
Comment (tpantelis) [inline tools/go.mod]: What is this? Haven't seen it before...

File: tools/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/submariner-io/shipyard/tools
 
-go 1.20
+go 1.21
```
Comment (tpantelis) [inline tools/go.mod]: Why was this bumped? Note that we specify 1.20 elsewhere.

Comment (dfarrell07) [reply tools/go.mod]: It seems like a dependency requires it. https://stackoverflow.com/questions/77079565/go-mod-tidy-will-auto-upgrade-go-version-in-go-mod

File: tools/go.mod (hunk 1 above)
Comment (dfarrell07) [reply tools/go.mod]: Yeah I'm not sure, I see https://go.dev/ref/mod#go-mod-file-toolchain

File: tools/go.mod (hunk 2 above)
Comment (tpantelis) [reply tools/go.mod]: Why wasn't this done/needed in devel? Or these CVE's haven't been applied there yet?

Comment (dfarrell07) [reply tools/go.mod]: Weird. If I use golang `go1.20.10` this doesn't happen. That seems to show the StackOverflow answer is wrong...

I'll try the set again with that go version.

**Comment by dfarrell07** [comment]
The commit message linter keeps globing this line together.

```
8: B1 Line exceeds max length (85>80): "go get -u go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
```

Continuing to try less-good ways.

**Comment by dfarrell07** [comment]
Even written like this, it fails. :exploding_head: 

```
Update opentelemetry to fix GHSA-rcjv-mgp8-qvmr

Update generated by

go get -u

go.opentelemetry.io/contrib/instrumentation/net/http/httptrace/otelhttptrace

go mod tidy
```

**Comment by dfarrell07** [comment]
[Clean up clusters](https://github.com/submariner-io/shipyard/actions/runs/6787139241/job/18449547246?pr=1465#logs) failed. I don't have an option to re-run it (yet?).

**Comment by dfarrell07** [comment]
> [Clean up clusters](https://github.com/submariner-io/shipyard/actions/runs/6787139241/job/18449547246?pr=1465#logs) failed. I don't have an option to re-run it (yet?).

I was able to re-run once the rest of the workflow completed. Waiting on results..

**Comment by dfarrell07** [comment]
> > [Clean up clusters](https://github.com/submariner-io/shipyard/actions/runs/6787139241/job/18449547246?pr=1465#logs) failed. I don't have an option to re-run it (yet?).
> 
> I was able to re-run once the rest of the workflow completed. Waiting on results..

And it passed :heavy_check_mark:

**Review by tpantelis — APPROVED** [review]

**Review by aswinsuryan — APPROVED** [review]

**Pull request merged**